include/bluetooth/bluetooth.h

```diff
@@ -71,6 +79,89 @@ int bt_enable(bt_ready_cb_t cb);
+ *  and store them in flash for later recovery, the way to do it would be
+ *  to first initialize the stack (using bt_enable), then call settings_load(),
+ *  and after that check with bt_id_get() how many identities were recovered.
+ *  If an insufficient amount of identities were recovered the app may then
+ *  call bt_id_create() to create new ones.
+ *
+ *  @param addr Address to use for the new identity. If NULL or initialized
+ *              to BT_ADDR_LE_ANY the stack will generate a new static
+ *              random address for the identity and copy it to the given
+ *              parameter upon return from this function (in case the
+ *              parameter was non-NULL).
+ *  @param irk  Identity Resolving Key (16 bytes) to be used with this
+ *              identity. If set to all zeroes or NULL, the stack will
+ *              generate a random IRK for the identity and copy it back
+ *              to the parameter upon return from this function (in case
+ *              the parameter was non-NULL). If privacy support
+ *              (CONFIG_BT_PRIVACY) is not enabled this parameter must
+ *              be NULL.
+ *
+ *  @return Identity identifier (>= 0) in case of success, or a negative
+ *          error code on failure.
+ */
+int bt_id_create(bt_addr_le_t *addr, u8_t *irk);
+
+/** @brief Reset/reclaim an identity for reuse.
+ *
+ *  The semantics of the @a addr and @a irk parameters of this function
+ *  are the same as with bt_id_create(). The difference is the first
+ *  @a id parameter that needs to be an existing identity (if it doesn't
+ *  exist this function will return an error). When given an existing
+ *  identity this function will disconnect any connections created using it,
+ *  remove any pairing keys or other data associated with it, and then create
+ *  a new identity in the same slot, based on the @a addr and @a irk
+ *  parameters.
+ *
+ *  @param id   Existing identity identifier.
+ *  @param addr Address to use for the new identity. If NULL or initialized
+ *              to BT_ADDR_LE_ANY the stack will generate a new static
+ *              random address for the identity and copy it to the given
+ *              parameter upon return from this function (in case the
+ *              parameter was non-NULL).
+ *  @param irk  Identity Resolving Key (16 bytes) to be used with this
+ *              identity. If set to all zeroes or NULL, the stack will
+ *              generate a random IRK for the identity and copy it back
+ *              to the parameter upon return from this function (in case
+ *              the parameter was non-NULL). If privacy support
+ *              (CONFIG_BT_PRIVACY) is not enabled this parameter must
+ *              be NULL.
+ *
+ *  @return Identity identifier (>= 0) in case of success, or a negative
+ *          error code on failure.
+ */
+int bt_id_reset(u8_t id, bt_addr_le_t *addr, u8_t *irk);
```

Comment:
Wouldn't the application anyway need to track unused identities, i.e. know what it's called `bt_id_clear()' for, so it doesn't go calling bt_le_adv_start() and triggering errors, or which identities it needs to call `bt_id_reset()` for before being able to use it again. So I'm not completely buying this argument. Another reason I'm reluctant to adding this is that it will make the implementation more complex and the storage less efficient in that there can be "holes" in the arrays that are stored. API-wise, this is not something that has an impact on what's in the current PR, so I hope this will not prevent proceeding with the rest.